src/microbe_stage/editor/MicrobeEditorReportComponent.cs

```diff
@@ -209,13 +212,27 @@ public void UpdateTimeIndicator(double value)
             .FormatSafe(Editor.CurrentGame.GameWorld.TotalPassedTime);
     }
 
-    public void UpdateGlucoseReduction(float value)
+    public void UpdateEvents(List<GameEventDescription> events)
     {
-        var percentage = Localization.Translate("PERCENTAGE_VALUE").FormatSafe(Math.Round(value * 100));
+        foreach (var currentLabel in majorEventsList.GetChildren())
+        {
+            if (currentLabel is Label && currentLabel != eventTemplate)
+            {
+                currentLabel.QueueFree();
+            }
+        }
 
-        // The amount of glucose has been reduced to {0} of the previous amount.
-        glucoseReductionLabel.Text = Localization.Translate("THE_AMOUNT_OF_GLUCOSE_HAS_BEEN_REDUCED")
-            .FormatSafe(percentage);
+        foreach (var registeredEvent in events)
+        {
+            if (registeredEvent.ShowInReport)
+            {
+                var duplicate = (Label)eventTemplate.Duplicate();
```

Comment:
Packed scenes should be used instead of duplicating nodes for instantiating things.